scripts/pre_ingest_sanitisation.py

```diff
@@ -0,0 +1,236 @@
+# For an annotated version, use the ipython notebook.
+#################################
+# Set these two variables, then rest should run:
+source_path = '/nfs/lsst2/photocalData/data/ctio0m9/'
+dest_root_path = '/nfs/lsst2/photocalData/data/ctio0m9_sanitised/'
+#################################
+
+
+
+import shutil
+import os
+import filecmp
+import pyfits as pf
+import sys
+import glob
+import functions as fn
+import lsst.daf.base as dafBase
+import re
+
+def SafeCopy(src,dest):
+    if os.path.exists(dest):
+        print 'warning, tried to overwrite %s with %s'%(dest, src)
+        if filecmp.cmp(src, dest):
+            print 'but it\'s OK, the files were the same anyway'
+        else:
+            print 'DISASTER - the files were different!\n\n\n'
+    else:
+        shutil.copy(src,dest)
+        
+def SafeMove(src,dest):
+    if os.path.exists(dest):
+        print 'warning, tried to overwrite %s with %s'%(dest, src)
+        if filecmp.cmp(src, dest):
+            print 'but it\'s OK, the files were the same anyway'
+        else:
+            print 'DISASTER - the files were different!\n\n\n'
+    else:
+        shutil.move(src,dest)      
+
+
+def GetFilename(filename_or_path):
+    return str(filename_or_path).split('/')[-1]
+
+if not os.path.exists(dest_root_path): os.mkdir(dest_root_path)
+
+print 'Copying data...'
+sys_call = 'cp -r %s* %s'%(source_path, dest_root_path)
+os.system(sys_call)
+print 'Data copied'
+
+os.mkdir(dest_root_path + 'one_amp')
+os.mkdir(dest_root_path + 'two_amp')
+os.mkdir(dest_root_path + 'wrong_size')
+os.mkdir(dest_root_path + 'bad_files')
+
+original_fileslist = []
+for path, subdirs, files in os.walk(dest_root_path):
+    for name in files:
+        if name.endswith('.fits'):
+            original_fileslist.append(os.path.join(path, name))
+
+
+bad_match = ['fitstest.fits', 'test.fits', 'ut161210.f075.fits']
+pop_list = []
+for filename in original_fileslist:
+    for _ in bad_match:
+        if filename.find(_)!=-1:
+            dest_path = dest_root_path + 'bad_files/'
+            dest_filename = dest_path + '_from_' +filename.replace('/','_') #filenames are not unique
+            SafeMove(filename, dest_filename)
+            pop_list.append(filename)
+            print "Moved: %s to \n%s"%(filename, dest_filename)
+            break # don't want to match twice per file
+            
+for popname in pop_list:
+    original_fileslist.remove(popname)
+
+
+headers = {}
+
+nfiles = len(original_fileslist)
+for i, filename in enumerate(original_fileslist):
+    hdulist = pf.open(filename)
+    main_header = hdulist[0].header
+    headers[filename] = main_header
+    if i%500==0: print 'loaded %s of %s'%(i, nfiles)
+
+print 'Finished loading headers'
```

Comment:
If you are sanitizing headers here too, is it reasonable to fix the dates here?

Oh I guess you don't want to modify the files?